pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarChannelInitializer.java

```diff
@@ -59,7 +59,8 @@ public class PulsarChannelInitializer extends ChannelInitializer<SocketChannel>
     // This cache is used to maintain a list of active connections to iterate over them
     // We keep weak references to have the cache to be auto cleaned up when the connections
     // objects are GCed.
-    private final Cache<SocketAddress, ServerCnx> connections = Caffeine.newBuilder()
+    // protected for test.
```

Comment:
you can add @VisibleForTesting annotation

Reply:
fine